src/Symfony/Component/Config/Tests/Definition/JsonSchemaGeneratorTest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Tests\Definition;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Config\Definition\ArrayNode;
+use Symfony\Component\Config\Definition\JsonSchemaGenerator;
+use Symfony\Component\Config\Tests\Fixtures\Configuration\ExampleConfiguration;
+
+class JsonSchemaGeneratorTest extends TestCase
+{
+    public function testExampleConfiguration()
+    {
+        $configuration = new ExampleConfiguration();
+
+        $tmp = tempnam(sys_get_temp_dir(), 'json-schema-generator');
+        register_shutdown_function(fn () => unlink($tmp));
```

Comment:
The cleanup of the file should be handled in `TestCase::tearDown`. Example in [`CacheWarmerTest`](https://github.com/symfony/symfony/blob/04551ad54245e0f48206eb05a582996b419c8c1e/src/Symfony/Component/HttpKernel/Tests/CacheWarmer/CacheWarmerTest.php#L21-L29)